src/cli.cr

```diff
@@ -76,6 +79,31 @@ module Shards
   end
 
   def self.build(path, args)
+    targets, options = parse_args(args)
+
+    begin
+      Commands::Check.run(path)
+    rescue
+      Commands::Install.run(path)
+    end
+
+    Commands::Build.run(path, targets, options)
+  end
+
+  def self.run_command(path, args)
+    targets, options = parse_args(args)
+
+    begin
+      Commands::Check.run(path)
+    rescue
+      Commands::Install.run(path)
+    end
+
+    Commands::Run.run(path, targets, options)
+  end
+end
+
+  def self.parse_args(args)
```

Comment:
It should stop parsing args when `--` is found, and return them as `run_options`, in order to pass args to the run executable, such as:

```console
$ shards run test --static -- --seed=abc name
# crystal build src/test.cr -o bin/test --static
# bin/test --seed=abc name
```

Reply:
This is now handled by passing an additional parameter into the run command for `run_options`.